### DISCUSSION TITLE
Availability of automatically generated badges

### DISCUSSION THREAD

**Discussion by goodboychan** [quarto-dev/quarto-cli#3937] [Q&A] [answered]
Hi,
Currently, I tried to move my contents from fastpages to quarto.
I just wonder that is it possible to generate badges for external link, such as colab or binder while converting notebook to html.
(I think it may be related to [this](https://github.com/trsvchn/colab-badge-action))

Thanks.

**Comment by mcanouil** [comment, marked as answer]
https://github.com/badges/shields provides an API to create whatever badge you want.

I must admit I am not sure to understand what badges, pointing to what external link and where you want to put them.

**Comment by goodboychan** [comment]
Thanks for the comment.

[Here](https://goodboychan.github.io/python/coursera/deeplearning.ai/2022/05/11/01-Logistic-Regression-with-a-Neural-Network.html) is my previous concept of badge arrangement. Badges are added just below the title.

Of course, I can add these manually. But it would be much easier if I can generate this with automatic processes (github actions, scripts whatever)

Sorry for poor description.

**Reply by mcanouil** [reply]
There aren't any badges created by CI/CD.
There are badges that reflect GitHub Actions workflows statuses but that's it.

Make your CI/CD, set the badges, add them where you like.

Sorry, but I really don't see what this has to do with Quarto CLI 🤷‍♂️

**Reply by goodboychan** [reply]
Got it, I didn't got any background about this process.

Thanks for your help.

**Comment by ianepreston** [comment]
I'm having the same issue as OP, maybe I can describe the desired behaviour a bit better. When creating a blog post with fastpages, you could add a `badges: true` element to the metadata at the top of the notebook. This would generate links at the top of the published post that would direct to open the notebook the post was generated from in either binder or Google collab. This in turn would allow users who wanted to alter the code in the notebook or otherwise experiment with it to interact with it. You can see an example of what this looks like from the [fastpages docs](https://fastpages.fast.ai/jupyter/2020/02/20/test.html) that also describes how to set it up.

While it would be possible to manually add this content to the top of notebooks, all the information necessary to generate them automatically should be available to quarto, so it would be a nice quality of life improvement.

I'd be willing to contribute to adding this feature if it would likely be accepted.

**Reply by mcanouil** [reply]
It feels like the following feature from 1.3 is what you are looking for in some aspects, see <https://quarto.org/docs/blog/posts/2023-03-15-multi-format/>.

Also note that you could do what your want by developing an extension to Quarto, see <https://quarto.org/docs/extensions/creating.html>.
